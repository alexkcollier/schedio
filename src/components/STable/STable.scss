.table {
  &__header {
    border-bottom: border();
    font-weight: $font-weight-bold;
    table-layout: auto;
    text-transform: uppercase;
  }

  &__footer {
    border-top: border();
    font-weight: $font-weight-bold;
  }

  &__caption {
    @include font-size(6);

    caption-side: bottom;
    color: color('night', 'light');
    font-weight: $font-weight-regular;
    text-align: left;

    &--top {
      caption-side: top;
    }
  }

  &__container,
  &__overlay {
    overflow-x: auto;
    position: relative;
  }

  &__overlay--left {
    &::before {
      @include shadow-overlay(to right);
    }
  }

  &__overlay--right {
    &::after {
      @include shadow-overlay(to left);
    }
  }

  // These row variants have been left here because they're dependent on `.table--<variant>`
  &--bordered {
    .table__body {
      .table__row {
        border-top: border();
      }
    }
  }

  &--hoverable {
    .table__body {
      .table__row {
        &:hover,
        &:focus {
          background-color: color('grey', 'light') !important;
        }
      }
    }
  }

  &--striped {
    .table__body {
      .table__row:nth-child(even) {
        background-color: lighten(color('grey', 'lighter'), 3%);
      }
    }
  }
}
