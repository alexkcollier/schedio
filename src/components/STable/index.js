import STable from './STable.vue'

export { STable }

export default { install: Vue => Vue.component(STable.name, STable) }
