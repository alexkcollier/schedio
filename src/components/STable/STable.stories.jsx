import { SHeading } from '@/components/SHeading';
import { STable } from '@/components/STable';
import { STableCell } from '@/components/STableCell';
import { STableRow } from '@/components/STableRow';
import PropList from '@@/docs/components/PropList';
import { boolean, number, object, text } from '@storybook/addon-knobs';

export default {
  title: 'Components/Tables/STable',

  parameters: {
    jest: 'STable',
  },
};

export const table = () => {
  const makeOptionalBoolean = (prop, val = false) => boolean(prop, val, 'Optional Props');
  const makeDummySlotData = (prop, n = 3) => number(prop, n, { range: true }, 'Slots');
  const slotExampleProps = {
    hasCaptionTop: makeOptionalBoolean('has-caption-top'),
    headerInFooter: makeOptionalBoolean('header-in-footer'),
    hideHeader: makeOptionalBoolean('hide-header'),
    isBordered: makeOptionalBoolean('is-bordered'),
    isHoverable: makeOptionalBoolean('is-hoverable'),
    isStriped: makeOptionalBoolean('is-striped'),
    isFullwidth: makeOptionalBoolean('is-fullwidth'),
  };
  const tableData = [
    { col1: 'Cell 1', col2: 'Cell 2', col3: 'Cell 3' },
    { col1: 'Cell 4', col2: 'Cell 5', col3: 'Cell 6' },
  ];
  const propExampleProps = {
    ...slotExampleProps,
    caption: text('caption', 'This is a caption', 'Optional Props'),
    tableData: object('table-data', tableData, 'Optional Props'),
  };

  return {
    props: {
      props: {
        default: () => ({
          slotExampleProps,
          propExampleProps,
        }),
      },
      headerCells: { default: makeDummySlotData('header (cells)') },
      bodyRows: { default: makeDummySlotData('default (rows)') },
      footerRows: { default: makeDummySlotData('footer (rows)', 1) },
    },
    render (h) {
      const { props, headerCells, bodyRows, footerRows } = this.$props;

      return (
        <div>
          <SHeading level="1">Table</SHeading>
          <p>Tables organize display data. They allow data to be viewed in detail.</p>

          <SHeading level="2">Examples</SHeading>
          <p>
            Tables can be implemented either with slots, or by passing an <code>{'<Array>'}</code>{' '}
            or <code>{'<Object>'}</code> to the <code>tableData</code> prop.
          </p>

          <SHeading level="3">Using props</SHeading>
          <p>
            When the <code>tableData</code> prop is provided, <code>{'<STable />'}</code> will
            automatically generate its content. If the items in <code>tableData</code> are not{' '}
            <code>{'<Array>'}</code>, the table's headers will be generated from those keys and
            automatically transformed to start case (e.g.: someHeader => 'Some Header'). You can
            hide the header with the <code>hideHeader</code> prop.
          </p>
          <p>You provide the following props to set manually their correspnding elements:</p>
          <ul>
            <li>
              <code>header</code>
            </li>
            <li>
              <code>footer</code>
            </li>
            <li>
              <code>caption</code>
            </li>
          </ul>
          <p>
            The header can be duplicated in the footer by setting <code>headerInFooter</code> to{' '}
            <code>true</code>. Note that <code>footer</code> will override this.
          </p>
          <p>
            In the table body, values of type <code>{'<Number>'}</code> will have{' '}
            <code>is-numeric</code> applied to their cell automatically.
          </p>
          <STable {...{ props: props.propExampleProps }} />

          <SHeading level="3">Using slots</SHeading>
          <p>
            When using slots, it is up to you to manually provide and configure{' '}
            <code>{'<STableRow />'}</code> and <code>{'<STableCell />'}</code>.
          </p>
          <p>Four slots are provided:</p>
          <ul>
            <li>
              <code>caption</code> &mdash; this will place content in <code>{'<caption />'}</code>
            </li>
            <li>
              <code>header</code> &mdash; this will place content in <code>{'<thead />'}</code>
            </li>
            <li>
              <code>default</code> &mdash; this will place content in <code>{'<tbody />'}</code>
            </li>
            <li>
              <code>footer</code> &mdash; this will place content in <code>{'<tfoot />'}</code>
            </li>
          </ul>

          <STable {...{ props: props.slotExampleProps }}>
            <template slot="caption">{props.propExampleProps.caption}</template>

            <STableRow slot="header">
              {Array.from({ length: headerCells }, (cell, idx) => (
                <STableCell key={`header-cell-${idx}`}>Cell {idx + 1}</STableCell>
              ))}
            </STableRow>

            {Array.from({ length: bodyRows }, (cell, idx) => (
              <STableRow key={`row-${idx}`}>
                {Array.from({ length: headerCells }, (cell, idx) => (
                  <STableCell key={`cell-${idx}`}>Cell {idx + 1}</STableCell>
                ))}
              </STableRow>
            ))}

            <template slot="footer">
              {Array.from({ length: footerRows }, (cell, idx) => (
                <STableRow key={`footer-row-${idx}`}>
                  {Array.from({ length: headerCells }, (cell, idx) => (
                    <STableCell key={`footer-cell-${idx}`}>Cell {idx + 1}</STableCell>
                  ))}
                </STableRow>
              ))}
            </template>
          </STable>

          <SHeading level="3">Responsive Tables</SHeading>
          <p>
            Tables are automatically responsive and scroll when they are too wide for their parent.
            An overlay will be added to either side that has hidden content.
          </p>

          <STable>
            <template slot="header">
              <STableRow>
                <STableCell>Narrow</STableCell>
                <STableCell>Very wide</STableCell>
              </STableRow>
            </template>
            <STableRow>
              <STableCell>This is narrow.</STableCell>
              <STableCell style="min-width: 100vw; background-color: whitesmoke;">
                This is very wide.
              </STableCell>
            </STableRow>
          </STable>

          <PropList component={STable} />
        </div>
      );
    },
  };
};

table.story = {
  name: 'Table',
};
