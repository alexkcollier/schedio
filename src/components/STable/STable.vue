<template>
  <div
    class="table__overlay"
    :class="overlayClassList"
  >
    <div class="table__container">
      <table
        v-bind="$attrs"
        :class="tableClassList"
        class="table"
      >
        <caption
          v-if="$slots.caption || caption"
          :class="{ 'table__caption--top': hasCaptionTop }"
          class="table__caption"
        >
          <slot name="caption">
            {{ caption }}
          </slot>
        </caption>

        <thead
          v-if="!hideHeader && ($slots.header || headerCells.length)"
          class="table__header"
        >
          <slot name="header">
            <STableRow>
              <STableCell
                v-for="(cell, key) in headerCells"
                :key="key"
              >
                {{ cell.value || cell | startCase }}
              </STableCell>
            </STableRow>
          </slot>
        </thead>

        <tbody class="table__body">
          <slot>
            <STableRow
              v-for="(row, key) in tableData"
              :key="key"
            >
              <STableCell
                v-for="cell in row"
                :key="cell"
                :is-header="cell.header"
                :is-numeric="typeof cell === 'number'"
              >
                {{ cell.value || cell }}
              </STableCell>
            </STableRow>
          </slot>
        </tbody>

        <tfoot
          v-if="$slots.footer || footerCells.length"
          class="table__footer"
        >
          <slot name="footer">
            <STableRow>
              <STableCell
                v-for="(cell, key) in footerCells"
                :key="key"
              >
                {{ cell.value || cell | startCase }}
              </STableCell>
            </STableRow>
          </slot>
        </tfoot>
      </table>
    </div>
  </div>
</template>

<script>
import startCase from 'lodash.startcase'
import { STableRow } from '@/components/STableRow'
import { STableCell } from '@/components/STableCell'

export default {
  name: 'STable',

  components: {
    STableRow,
    STableCell,
  },

  filters: {
    startCase,
  },

  props: {
    tableData: {
      type: [Array, Object],
      default: () => [],
    },
    header: {
      type: [Array, Object],
      default: () => [],
    },
    footer: {
      type: [Array, Object],
      default: () => [],
    },
    caption: {
      type: String,
      default: '',
    },
    hasCaptionTop: {
      type: Boolean,
      default: false,
    },
    headerInFooter: {
      type: Boolean,
      default: false,
    },
    hideHeader: {
      type: Boolean,
      default: false,
    },
    isBordered: {
      type: Boolean,
      default: false,
    },
    isHoverable: {
      type: Boolean,
      default: false,
    },
    isStriped: {
      type: Boolean,
      default: false,
    },
    isFullwidth: {
      type: Boolean,
      default: false,
    },
  },

  data () {
    return {
      container: null,
      containerWidth: 0,
      tableWidth: 0,
      distanceToScroll: 0,
      horizontalScrollPosition: 0,
    }
  },

  computed: {
    overlayClassList () {
      return {
        'table__overlay--left': this.isScrolled,
        'table__overlay--right': this.isFullyScrolled,
      }
    },

    tableClassList () {
      return {
        'table--bordered': this.isBordered,
        'table--hoverable': this.isHoverable,
        'table--striped': this.isStriped,
        'table--fullwidth': this.isFullwidth,
      }
    },

    isScrolled () {
      return this.horizontalScrollPosition > 0
    },

    isFullyScrolled () {
      return this.horizontalScrollPosition < this.distanceToScroll
    },

    headerCells () {
      // check keys in case object is passed
      if (Object.keys(this.header).length) return this.header

      const items = Array.isArray(this.tableData) ? this.tableData : Object.values(this.tableData)
      const headers = items.reduce((acc, item) => {
        return Array.isArray(item) ? acc : acc.concat(Object.keys(item))
      }, [])

      // only return unique values
      return [...new Set(headers)]
    },

    footerCells () {
      // give `this.footer` priority
      if (Object.keys(this.footer).length) return this.footer
      // always return something since we only check for this.footerCells.length
      return this.headerInFooter ? this.headerCells : this.footer
    },
  },

  created () {
    // warn if `tableData` prop or default slot not provided.
    // check keys in case object is passed
    if (!Object.keys(this.tableData).length && !this.$slots.default) {
      console.error(`\`${this.$options.name}\` requires \`tableData\` prop or default slot content`)
    }
  },

  mounted () {
    this.container = this.$el.querySelector('.table__container')
    this.checkForOverflow()

    if (typeof window !== 'undefined') {
      window.addEventListener('resize', this.checkForOverflow)
      this.container.addEventListener('scroll', this.checkOverflowSide)
    }
  },

  destroyed () {
    if (typeof window !== 'undefined') {
      window.removeEventListener('resize', this.checkForOverflow)
      this.container.removeEventListener('scroll', this.checkOverflowSide)
    }
  },

  methods: {
    checkForOverflow (e) {
      const table = this.$el.querySelector('.table')

      this.containerWidth = this.container.clientWidth
      this.tableWidth = table.scrollWidth
      this.distanceToScroll = this.tableWidth - this.containerWidth
      this.horizontalScrollPosition = table.scrollLeft
    },

    checkOverflowSide (e) {
      this.horizontalScrollPosition = e.target.scrollLeft
    },
  },
}
</script>

<style lang="scss">
.table {
  &__header {
    border-bottom: border();
    font-weight: font-weight('bold');
    table-layout: auto;
    text-transform: uppercase;
  }

  &__footer {
    border-top: border();
    font-weight: font-weight('bold');
  }

  &__caption {
    @include font-size(6);

    caption-side: bottom;
    color: color('night', 'light');
    font-weight: font-weight('regular');
    text-align: left;

    &--top {
      caption-side: top;
    }
  }

  &__container,
  &__overlay {
    overflow-x: auto;
    position: relative;
  }

  &__overlay--left {
    &::before {
      @include shadow-overlay(to right);
    }
  }

  &__overlay--right {
    &::after {
      @include shadow-overlay(to left);
    }
  }

  // These row variants have been left here because they're dependent on `.table--<variant>`
  &--bordered {
    .table__body {
      .table__row {
        border-top: border();
      }
    }
  }

  &--hoverable {
    .table__body {
      .table__row {
        &:hover,
        &:focus {
          background-color: color('grey', 'light') !important;
        }
      }
    }
  }

  &--striped {
    .table__body {
      .table__row:nth-child(even) {
        background-color: lighten(color('grey', 'lighter'), 3%);
      }
    }
  }

  &--fullwidth {
    min-width: 100%;
  }
}
</style>
