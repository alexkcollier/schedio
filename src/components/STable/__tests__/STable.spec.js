import { mount } from '@vue/test-utils'
import STable from '@/components/STable/STable.vue'
import { axe, toHaveNoViolations } from 'jest-axe'

expect.extend(toHaveNoViolations)

describe('STable.vue', () => {
  const errorSpy = jest.spyOn(global.console, 'error').mockImplementation(() => {})
  let wrapper

  beforeEach(() => {
    wrapper = mount(STable, {
      propsData: {
        tableData: [
          {
            col1: 'Cell 1',
            col2: 'Cell 2',
            col3: 'Cell 3',
          },
          {
            col1: 'Cell 4',
            col2: 'Cell 5',
            col3: 'Cell 6',
          },
        ],
      },
    })
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  it('renders correctly', async () => {
    expect(wrapper.html()).toMatchSnapshot()
    expect(await axe(wrapper.html())).toHaveNoViolations()
  })

  it('requires data', () => {
    mount(STable)
    expect(errorSpy).toHaveBeenCalled()
  })

  it('allows slots to be used for data', () => {
    const slotWrapper = mount(STable, {
      slots: {
        caption: '<span>Caption</span>',
        header: '<tr><th>Header</th></tr>',
        default: '<tr><td>Slot content</td></tr>',
        footer: '<tr><td>Footer</td></tr>',
      },
    })

    expect(errorSpy).not.toHaveBeenCalled()
    expect(slotWrapper.contains('caption')).toBe(true)
    expect(slotWrapper.contains('tbody')).toBe(true)
    expect(slotWrapper.contains('thead')).toBe(true)
    expect(slotWrapper.contains('tfoot')).toBe(true)

    expect(slotWrapper.html()).toMatchSnapshot()
  })

  it('generates header cells from data', () => {
    expect(wrapper.contains('th')).toBe(true)
  })

  it('can override generated headers', () => {
    wrapper.setProps({
      header: [
        'One',
        'Two',
        'Three',
      ],
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('can hide header cells', () => {
    wrapper.setProps({ hideHeader: true })
    expect(wrapper.contains('thead')).toBe(false)
  })

  it('can duplicate the header in the footer', () => {
    wrapper.setProps({ headerInFooter: true })
    const headerCells = wrapper.findAll('th')
    const footerCells = wrapper.findAll('tfoot > tr > td')

    expect(footerCells.length).toBe(headerCells.length)

    for (let i = 0; i < headerCells.length; i++) {
      expect(footerCells.at(i).text()).toBe(headerCells.at(i).text())
    }

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('can have a caption', () => {
    wrapper.setProps({ caption: 'caption' })
    expect(wrapper.contains('caption')).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('can have the caption at the top', () => {
    wrapper.setProps({ hasCaptionTop: true, caption: 'caption' })
    expect(wrapper.contains('.table__caption--top')).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('can be bordered', () => {
    wrapper.setProps({ isBordered: true })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('can be hoverable', () => {
    wrapper.setProps({ isHoverable: true })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('can be striped', () => {
    wrapper.setProps({ isStriped: true })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
