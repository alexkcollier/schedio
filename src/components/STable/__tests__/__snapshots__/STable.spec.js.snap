// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`STable.vue allows slots to be used for data 1`] = `
<div class="table__overlay">
  <div class="table__container">
    <table class="table">
      <caption class="table__caption"><span>Caption</span></caption>
      <thead class="table__header">
        <tr>
          <th>Header</th>
        </tr>
      </thead>
      <tbody class="table__body">
        <tr>
          <td>Slot content</td>
        </tr>
      </tbody>
      <tfoot class="table__footer">
        <tr>
          <td>Footer</td>
        </tr>
      </tfoot>
    </table>
  </div>
</div>
`;

exports[`STable.vue can be bordered 1`] = `
<div class="table__overlay">
  <div class="table__container">
    <table class="table table--bordered">
      <!---->
      <thead class="table__header">
        <tr class="table__row">
          <th class="table__cell">
            Col 1
          </th>
          <th class="table__cell">
            Col 2
          </th>
          <th class="table__cell">
            Col 3
          </th>
        </tr>
      </thead>
      <tbody class="table__body">
        <tr class="table__row">
          <td class="table__cell">
            Cell 1
          </td>
          <td class="table__cell">
            Cell 2
          </td>
          <td class="table__cell">
            Cell 3
          </td>
        </tr>
        <tr class="table__row">
          <td class="table__cell">
            Cell 4
          </td>
          <td class="table__cell">
            Cell 5
          </td>
          <td class="table__cell">
            Cell 6
          </td>
        </tr>
      </tbody>
      <!---->
    </table>
  </div>
</div>
`;

exports[`STable.vue can be fullwidth 1`] = `
<div class="table__overlay">
  <div class="table__container">
    <table class="table table--fullwidth">
      <!---->
      <thead class="table__header">
        <tr class="table__row">
          <th class="table__cell">
            Col 1
          </th>
          <th class="table__cell">
            Col 2
          </th>
          <th class="table__cell">
            Col 3
          </th>
        </tr>
      </thead>
      <tbody class="table__body">
        <tr class="table__row">
          <td class="table__cell">
            Cell 1
          </td>
          <td class="table__cell">
            Cell 2
          </td>
          <td class="table__cell">
            Cell 3
          </td>
        </tr>
        <tr class="table__row">
          <td class="table__cell">
            Cell 4
          </td>
          <td class="table__cell">
            Cell 5
          </td>
          <td class="table__cell">
            Cell 6
          </td>
        </tr>
      </tbody>
      <!---->
    </table>
  </div>
</div>
`;

exports[`STable.vue can be hoverable 1`] = `
<div class="table__overlay">
  <div class="table__container">
    <table class="table table--hoverable">
      <!---->
      <thead class="table__header">
        <tr class="table__row">
          <th class="table__cell">
            Col 1
          </th>
          <th class="table__cell">
            Col 2
          </th>
          <th class="table__cell">
            Col 3
          </th>
        </tr>
      </thead>
      <tbody class="table__body">
        <tr class="table__row">
          <td class="table__cell">
            Cell 1
          </td>
          <td class="table__cell">
            Cell 2
          </td>
          <td class="table__cell">
            Cell 3
          </td>
        </tr>
        <tr class="table__row">
          <td class="table__cell">
            Cell 4
          </td>
          <td class="table__cell">
            Cell 5
          </td>
          <td class="table__cell">
            Cell 6
          </td>
        </tr>
      </tbody>
      <!---->
    </table>
  </div>
</div>
`;

exports[`STable.vue can be striped 1`] = `
<div class="table__overlay">
  <div class="table__container">
    <table class="table table--striped">
      <!---->
      <thead class="table__header">
        <tr class="table__row">
          <th class="table__cell">
            Col 1
          </th>
          <th class="table__cell">
            Col 2
          </th>
          <th class="table__cell">
            Col 3
          </th>
        </tr>
      </thead>
      <tbody class="table__body">
        <tr class="table__row">
          <td class="table__cell">
            Cell 1
          </td>
          <td class="table__cell">
            Cell 2
          </td>
          <td class="table__cell">
            Cell 3
          </td>
        </tr>
        <tr class="table__row">
          <td class="table__cell">
            Cell 4
          </td>
          <td class="table__cell">
            Cell 5
          </td>
          <td class="table__cell">
            Cell 6
          </td>
        </tr>
      </tbody>
      <!---->
    </table>
  </div>
</div>
`;

exports[`STable.vue can duplicate the header in the footer 1`] = `
<div class="table__overlay">
  <div class="table__container">
    <table class="table">
      <!---->
      <thead class="table__header">
        <tr class="table__row">
          <th class="table__cell">
            Col 1
          </th>
          <th class="table__cell">
            Col 2
          </th>
          <th class="table__cell">
            Col 3
          </th>
        </tr>
      </thead>
      <tbody class="table__body">
        <tr class="table__row">
          <td class="table__cell">
            Cell 1
          </td>
          <td class="table__cell">
            Cell 2
          </td>
          <td class="table__cell">
            Cell 3
          </td>
        </tr>
        <tr class="table__row">
          <td class="table__cell">
            Cell 4
          </td>
          <td class="table__cell">
            Cell 5
          </td>
          <td class="table__cell">
            Cell 6
          </td>
        </tr>
      </tbody>
      <tfoot class="table__footer">
        <tr class="table__row">
          <td class="table__cell">
            Col 1
          </td>
          <td class="table__cell">
            Col 2
          </td>
          <td class="table__cell">
            Col 3
          </td>
        </tr>
      </tfoot>
    </table>
  </div>
</div>
`;

exports[`STable.vue can have a caption 1`] = `
<div class="table__overlay">
  <div class="table__container">
    <table class="table">
      <caption class="table__caption">
        caption
      </caption>
      <thead class="table__header">
        <tr class="table__row">
          <th class="table__cell">
            Col 1
          </th>
          <th class="table__cell">
            Col 2
          </th>
          <th class="table__cell">
            Col 3
          </th>
        </tr>
      </thead>
      <tbody class="table__body">
        <tr class="table__row">
          <td class="table__cell">
            Cell 1
          </td>
          <td class="table__cell">
            Cell 2
          </td>
          <td class="table__cell">
            Cell 3
          </td>
        </tr>
        <tr class="table__row">
          <td class="table__cell">
            Cell 4
          </td>
          <td class="table__cell">
            Cell 5
          </td>
          <td class="table__cell">
            Cell 6
          </td>
        </tr>
      </tbody>
      <!---->
    </table>
  </div>
</div>
`;

exports[`STable.vue can have the caption at the top 1`] = `
<div class="table__overlay">
  <div class="table__container">
    <table class="table">
      <caption class="table__caption table__caption--top">
        caption
      </caption>
      <thead class="table__header">
        <tr class="table__row">
          <th class="table__cell">
            Col 1
          </th>
          <th class="table__cell">
            Col 2
          </th>
          <th class="table__cell">
            Col 3
          </th>
        </tr>
      </thead>
      <tbody class="table__body">
        <tr class="table__row">
          <td class="table__cell">
            Cell 1
          </td>
          <td class="table__cell">
            Cell 2
          </td>
          <td class="table__cell">
            Cell 3
          </td>
        </tr>
        <tr class="table__row">
          <td class="table__cell">
            Cell 4
          </td>
          <td class="table__cell">
            Cell 5
          </td>
          <td class="table__cell">
            Cell 6
          </td>
        </tr>
      </tbody>
      <!---->
    </table>
  </div>
</div>
`;

exports[`STable.vue can override generated headers 1`] = `
<div class="table__overlay">
  <div class="table__container">
    <table class="table">
      <!---->
      <thead class="table__header">
        <tr class="table__row">
          <th class="table__cell">
            One
          </th>
          <th class="table__cell">
            Two
          </th>
          <th class="table__cell">
            Three
          </th>
        </tr>
      </thead>
      <tbody class="table__body">
        <tr class="table__row">
          <td class="table__cell">
            Cell 1
          </td>
          <td class="table__cell">
            Cell 2
          </td>
          <td class="table__cell">
            Cell 3
          </td>
        </tr>
        <tr class="table__row">
          <td class="table__cell">
            Cell 4
          </td>
          <td class="table__cell">
            Cell 5
          </td>
          <td class="table__cell">
            Cell 6
          </td>
        </tr>
      </tbody>
      <!---->
    </table>
  </div>
</div>
`;

exports[`STable.vue renders correctly 1`] = `
<div class="table__overlay">
  <div class="table__container">
    <table class="table">
      <!---->
      <thead class="table__header">
        <tr class="table__row">
          <th class="table__cell">
            Col 1
          </th>
          <th class="table__cell">
            Col 2
          </th>
          <th class="table__cell">
            Col 3
          </th>
        </tr>
      </thead>
      <tbody class="table__body">
        <tr class="table__row">
          <td class="table__cell">
            Cell 1
          </td>
          <td class="table__cell">
            Cell 2
          </td>
          <td class="table__cell">
            Cell 3
          </td>
        </tr>
        <tr class="table__row">
          <td class="table__cell">
            Cell 4
          </td>
          <td class="table__cell">
            Cell 5
          </td>
          <td class="table__cell">
            Cell 6
          </td>
        </tr>
      </tbody>
      <!---->
    </table>
  </div>
</div>
`;
